dvc/remote/base.py

```diff
@@ -912,20 +925,21 @@ def _cache_exists_traverse(
             ]
         with Tqdm(
             desc="Querying "
-            + ("cache in " + name if name else "remote cache"),
-            total=len(traverse_prefixes),
-            unit="dir",
+            + ("cache in '{}'".format(name) if name else "remote cache"),
+            total=remote_size,
+            initial=len(remote_checksums),
+            unit="objects",
         ) as pbar:
 
             def list_with_update(prefix):
-                ret = map(
+                return map(
                     self.path_to_checksum,
-                    list(self.list_cache_paths(prefix=prefix)),
-                )
-                pbar.update_desc(
-                    "Querying cache in '{}'".format(self.path_info / prefix)
+                    list(
+                        self.list_cache_paths(
+                            prefix=prefix, progress_callback=pbar.update
+                        )
+                    ),
```

Comment:
let's move `list` into a separate variable and then `map(self.path_to_checksum, paths)`. Just to make it easier to read :slightly_smiling_face: